plugin/fallback/fallback.go

```diff
@@ -0,0 +1,59 @@
+// Package fallback implements a fallback plugin for CoreDNS
+package fallback
+
+import (
+	"context"
+
+	"github.com/miekg/dns"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/nonwriter"
+	"github.com/coredns/coredns/plugin/proxy"
+)
+
+// Fallback plugin
+type Fallback struct {
+	Next  plugin.Handler
+	trace plugin.Handler
+	rules map[int]proxy.Upstream
+	proxy proxyCreator
+}
+
+// proxyCreator creates a proxy with the specified upstream
+type proxyCreator interface {
+	Create(trace plugin.Handler, upstream proxy.Upstream) plugin.Handler
```

Comment:
Typically called `New` in Go

Reply:
Why is this an interface? Only has one function so you can just use NewFunc with that signature?

Reply:
This allows me to have a stub implementation used for unit test.

Plus, will rename method as per earlier comment.